### ISSUE TITLE
get_rss_from_url fails at continue instruction

### ISSUE THREAD

**Issue by OpalSec** [vxunderground/ThreatIntelligenceDiscordBot#11]
Full disclosure - I don't have a coding background and have been figuring this out as I've gone, so apologies if the answer/error is obvious and I've missed it! Just looking to be pointed in the right direction if possible.

When running the script it appears to run through multiple feeds, but never prints anything to Discord despite there being relevant articles to fetch.

I added some print statements to try and figure out where it was getting lost - note the print("true") check immediately before the continue:

```
def get_rss_from_url(rss_item, hook_channel_descriptor):
    news_feed = feedparser.parse(rss_item[0])
    date_activity = None

    print("looking at " f'{rss_item[0]}')

    for rss_object in news_feed.entries:
        try:
            date_activity = time.strftime('%Y-%m-%dT%H:%M:%S', rss_object.published_parsed)
        except: 
            date_activity = time.strftime('%Y-%m-%dT%H:%M:%S', rss_object.updated_parsed)

        try:
            config_entry = config_file.get('main', rss_item[1])
        except NoOptionError: # automatically add newly discovered groups to config
            config_file.set('main', rss_item[1], " = ?")
            config_entry = config_file.get('main', rss_item[1])   

        if config_entry.endswith('?'):
            config_file.set('main', rss_item[1], date_activity)
        else:
            if(config_entry >= date_activity):
                print("true")
                continue
            else:
                config_file.set('main', rss_item[1], date_activity)

        print("passed time check")
```

This is the terminal output - it spams "true" but then skips the "passed time check" print statement and seems to jump straight to processing the next feed:

```
PS G:\GithubDesktop\Repos\ThreatIntelligenceDiscordBot\Source> python .\test_DiscordIntelBot.py
looking at https://grahamcluley.com/feed/
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
true
looking at https://threatpost.com/feed/
true
true
true
true
true
true
true
true
true
true
```

I don't understand why it's spammed the "true" print and skipped over the other print statement and jumped straight to the end. I thought the continue instruction should just tell it to run the next instruction?

**Comment by bertmad3400** [comment]
So this is actually the intended behavior of the bot. The function get_rss_from_url takes a single RSS feed (from a source like Bleeping Computer) and then processes all of the news articles from that feed. Hovewer, as the bot does this once every half an hour, it is necessary to implement a function to check if it has already sent the article, as the RSS feed doesn't update so often, which is what you se in action here.

Basically the bot writes the publish date of the latest article for each source that it scrapes to a central config file. Then, next time it's run, it goes through all the news articles in every feed, and only if the publish date is more recent than the one in the central config file, it will sent the article to discord and write the new latest publish date to the config file. If however it isn't more recent than the time in the config file,  it will simply skip the article using the "continue" statement, which is what you se when it spams true. Basically it's saying "this article has already been sent, skipping it"

Edit: Now, there is however a single bug in the bot, which is suspect might be a part of your problem. The bot first scrapes the articles and writes the latest publish date to the config file (both which doesn't take very long) and then it sents the relevant articles to discord (a proccess that takes a lot longer). The problem with this is that if you run the bot for a few minutes, stop execution and then run it again, what will happen is in the first run it will collect the articles, write the latest publish date to the config file, but then not get time for sending the articles to discord, and so, next time it runs, it will read the central config file, conclude it has already sent all the articles (which it haven't) and simply skip them. This bug has already been addressed in #10 (which includes some major rewrites of the code you're currently looking at), but that has yet to be merged to main.

**Comment by vxunderground** [comment]
Please update to latest instance.
